client/signup/steps/domains/index.jsx

```diff
@@ -690,23 +663,22 @@ class DomainsStep extends Component {
 		return this.props.isDomainOnly ? 'domain-first' : 'signup';
 	}
 
+	resetState() {
+		if ( inputMode.domainInput === this.state?.step ) {
+			this.setState( {} );
+		}
+	}
+
 	renderContent() {
 		let content;
 		let sideContent;
 
-		if ( 'mapping' === this.props.stepSectionName ) {
-			content = this.mappingForm();
-		}
-
-		if ( 'transfer' === this.props.stepSectionName ) {
-			content = this.transferForm();
-		}
-
 		if ( 'use-your-domain' === this.props.stepSectionName ) {
 			content = this.useYourDomainForm();
 		}
 
 		if ( ! this.props.stepSectionName || this.props.isDomainOnly ) {
+			this.resetState();
```

Comment:
Rendering is not a good place to perform any side effects or set state.

Reply:
@tyxla Thanks for catching that! This code was for testing purposes (only), and I forgot to remove it. 
It's not doing anything currently. I'll create a PR for it 😄